src/main/java/net/sacredlabyrinth/phaed/simpleclans/hooks/DiscordHook.java

```diff
@@ -334,32 +334,42 @@ public boolean categoryExists(String categoryId) {
     /**
      * Creates a new {@link ClanPlayer.Channel} in defined {@link Category}
      *
-     * <p>Sets {@link Permission#VIEW_CHANNEL} permission to all linked clan members.</p>
+     * <p>Sets positive {@link Permission#VIEW_CHANNEL} permission to all linked clan members.</p>
      *
-     * @return true, if channel was created. False, if category reached the limit.
+     * @throws InvalidChannelException clan is not verified or permanent, no one member is linked or clan is not in the whitelist.
+     * @throws LimitReachedException   if category reached the limit.
      */
-    public boolean createChannel(@NotNull Category category, @NotNull String clanTag) {
+    public void createChannel(@NotNull Category category, @NotNull String clanTag) throws InvalidChannelException, LimitReachedException {
         if (category.getTextChannels().size() >= MAX_CHANNELS_PER_CATEGORY) {
-            return false;
+            throw new LimitReachedException();
```

Comment:
We probably need two exceptions: ChannelLimit and CategoryLimit

If this one is thrown, the loop for creating the channels will break.

Reply:
And what about other exceptions? 🤔

Reply:
What do you mean?

Reply:
I think, we can create 3 exceptions: `ChannelLimitException`, `CategoryLimitException` and `InvalidChannelException`.
`ChannelLimitException` and `InvalidChannelException` will be ignored.

Reply:
Wait
ChannelLimit is unnecessary, we are already checking it here:
https://github.com/RoinujNosde/SimpleClans/pull/279/files#diff-374b767dba884d240f5320f5da8258d2f3d7b00c3afd212cfa3e1b8824547668R308-R309

Reply:
Not exactly. We have 2 methods: `createChannel(Category, clanTag)` and `createChannel(clanTag)`.
Let's imagine, I'm the guy who uses SC API and what I'll see in case where we won't do *CategoryLimitException*?
I'll see that only overloaded method (`createChannel(clanTag)`) will take care of categories limit.

Reply:
We should handle the channel limit internally by trying to create it in a different category, that's why the exception ChannelLimit shouldn't be available to the API users.

Now the CategoryLimit is something we can't handle, so we need to let API users know.

Reply:
Why we can't handle `CategoryLimit`? 👀 
There is a [place ](https://github.com/RoinujNosde/SimpleClans/pull/279/files/4c881f4601812c9c7c38c6a6d553fd19b63f8400#diff-374b767dba884d240f5320f5da8258d2f3d7b00c3afd212cfa3e1b8824547668R314)where it throws.

Reply:
I mean there is nothing we can do when the category limit is reached, we can't increase it, it's limited by discord.

Reply:
> I mean there is nothing we can do when the category limit is reached, we can't increase it, it's limited by discord.

Sure, but what if I want to make an automatic giveaway when channels are fulled? 😃 
We can't know where it will be useful.

Reply:
Alright then